doc/sphinx/src/models.rst

```diff
@@ -1491,16 +1492,114 @@ key-value pairs. For exampe the following input deck is for air:
+
+.. code-block::
+
+  # defaults are true
+  piecewiseRho = false
+  piecewiseT = false
+  piecewiseSie = dalse
+
+These options may be true or false. The default is true. When
+piecewise grids are active, the density-temperature (or
+density-energy) grid is built as a Cartesian product grid of grids of
+non-uniform resolutions. The density grid gets split into three
+pieces, a region ``[rhoMin, rhoFineMin]``, a region ``[rhoFineMin,
+rhoFineMax]``, and a region ``[rhoFineMin, rhoMax]``. The
+``numrho/decade`` parameter sets the number of points per decade in
+the central refined region. The regions at lower and higher density
+have ``rhoCoarseFactorLo`` and ``rhoCoarseFactorHi`` fewer points per
+decade respectively compared to the finer region.
+
+Typically the fine region should be roughly centered around the normal
+density for a material, which is usually a challenging region to
+capture. If you neglect to set ``rhoFineMin`` and ``rhoFineMax``,
+``sesame2spiner`` will set the central refined region to be a region
+of diameter ``rhoFineDiameterDecades`` (in log space) around the
+material's normal density.
+
+The temperature grid has two regions, a more finely spaced region at
+low temperatures and a less finely spaced region at high
+temperatures. The regions are spearated by a temperature
+``TSplitPoint``. The default is :math:`10^4` Kelvin. The energy grid
+follows the temperature grid, with the energy split point
+corresponding to the temperature split point. The coarser
+high-temperature temperature and energy grids are coarsened by a
+factor of ``TCoarseFactor`` and ``sieCoarseFactor`` respectively.
+
+A diagram of a density-temperature grid is shown below. The region
+with temperatures below ``TSplitPoint`` is refined in temperature. The
+region between ``rhoFineMin`` and ``rhoFineMax`` is refined in
+density.
+
+.. image:: phase_diagram.png
+  :width: 400
+  :alt: An example piecewise density-temperature grid.
+
+
+Thus the input block for piecewise grid might look like this:
+
+.. code-block::
+
+  # Below, all right-hand-sides are set to their default values.
+  piecewiseRho = true
+  piecewiseT = true
+  piecewiseSie = true
+
+  # the fine resolution for rho.
+  numrho/decade = 350
+  # width of the fine region for rho
+  rhoFineDiameterDecades = 1.5
+  # the lower density region is 3x less refined
+  rhoCoarseFactorLo = 3
+  # the higher density region is 5x less refined
+  rhoCoarseFactorHi = 5
+
+  # the fine resolution for T
+  numT/decade = 100
+  # the point demarking the coarse and fine regions in temperature
+  TSplitPoint = 1e4
+  # it's usually wise to to not let
+  # temperature get too small in log space if you do this
+  Tmin = 1
+  # The coarser region (above the split point) is 50 percent less refined
+  TCoarseFactor = 1.5
+
+  # energy has the split point sie(rhonormal, TSplitPoint)
+  # but we may still specify the resolution
+  numSie/decade = 100
+  sieCoarseFactor = 1.5
```

Comment:
So I take it that if for example we want to increase to say 10 grid regions in some dimension, we'll need to rework how this input syntax works, yes?

Reply:
I wrote this at the beginning of the review. The answer is yes, but I think this is a good step in the right direction.